src/window_keyboard.cpp

```diff
@@ -237,42 +238,37 @@ void Window_Keyboard::Refresh() {
 void Window_Keyboard::Update() {
 	Window_Base::Update();
 
+	// move left on wide fields
+	int skip_dir = -1;
+
 	if (active) {
 		if (Input::IsRepeated(Input::DOWN)) {
 			play_cursor = true;
 			row = (row + 1) % row_max;
-
-			if (col > 0 && GetSelected().empty() && !GetKey(row, col - 1).empty()) {
-				col--;
-			}
 		}
 		if (Input::IsRepeated(Input::UP)) {
 			play_cursor = true;
 			row = (row + row_max - 1) % row_max;
-
-			if (col > 0 && GetSelected().empty() && !GetKey(row, col - 1).empty()) {
-				col--;
-			}
 		}
 		if (Input::IsRepeated(Input::RIGHT)) {
 			play_cursor = true;
-			col += 1;
-			if (col >= col_max) {
-				col = 0;
-				if (mode == Letter) { row = (row + 1) % row_max; }
-			}
+			col = (col + 1) % col_max;
+			skip_dir = 1;
 		}
 		if (Input::IsRepeated(Input::LEFT)) {
 			play_cursor = true;
-			col -= 1;
-			if (col < 0) {
-				col = col_max - 1;
-				if (mode == Letter) { row = (row + row_max - 1) % row_max; }
-			}
+			col = (col + col_max - 1) % col_max;
 		}
+	}
 
+	// Special handling for wide fields
+	if (col > 0 && col < col_max) {
+		std::string k = GetKey(row, col - 1);
+		if (k[0] == '<' || k == SPACE)
+			col = std::min(col + skip_dir, col_max - 1);
```

Comment:
Is it possible check for SYMBOL and DONE based on the row-col position instead?
Imho checking for "<" is a bit ugly.

Reply:
Yeah, my intention was here to not have to compare with all the different DONEs. But you are right, as long as they stay on their position your idea is good.